### ISSUE TITLE
child workflow not executed

### ISSUE THREAD

**Issue by 237779932** [elsa-workflows/elsa-core#4768]
```
// Setup service container.
var services = new ServiceCollection();

// Add Elsa services.
services.AddElsa();

// Build service container.
var serviceProvider = services.BuildServiceProvider();

// Resolve a workflow runner to run the workflow.
var workflowRunner = serviceProvider.GetRequiredService<IWorkflowRunner>();

// Run the workflow.
var result = await workflowRunner.RunAsync<ParentWorkflow>();




public class ParentWorkflow : WorkflowBase
{
    protected override void Build(IWorkflowBuilder builder)
    {
        var childOutput = builder.WithVariable<IDictionary<string, object>>();

        builder.Root = new Sequence
        {
            Activities =
            {
                new WriteLine("I am the parent workflow."),
                new DispatchWorkflow
                {
                    WorkflowDefinitionId = new(nameof(ChildWorkflow)),
                    Input = new(new Dictionary<string, object>
                    {
                        ["ParentMessage"] = "Hello from parent!"
                    }),
                    WaitForCompletion = new(true),
                    Result = new(childOutput)
                },
                new WriteLine(context => $"Child finished executing and said: {childOutput.Get(context)!["ChildMessage"]}")
            }
        };
    }
}

public class ChildWorkflow : WorkflowBase
{
    protected override void Build(IWorkflowBuilder builder)
    {
        builder.Root = new Sequence
        {
            Activities =
            {
                new WriteLine(context => $"I am the child workflow. My parent said: \"{context.GetInput<string>("ParentMessage")}\"."),
                new WriteLine("Let's say hi to the parent!"),
                new SetOutput
                {
                    OutputName = new("ChildMessage"),
                    OutputValue = new("Hello from child!")
                }
            }
        };
    }
}

```

What am I supposed to do now ?

**Comment by sfmskywalker** [comment]
Hi @237779932,

Your issue stems from the DispatchWorkflow activity's reliance on hosted services, which are integral to an application host's functionality. In a Console application without such a host, this feature won't operate as expected.

To resolve this, you'll need to establish an ASP.NET Core Web application or a Worker Service application. These environments provide the necessary infrastructure for DispatchWorkflow to function correctly. You can find detailed instructions for setting up an ASP.NET Core web application with Elsa [here](https://v3.elsaworkflows.io/docs/installation/elsa-web).

While these instructions include installing the Controllers service, it's not mandatory for your purpose. The crucial steps are creating a WebApplication object and configuring Elsa to suit your needs. If you encounter any specific issues during setup or configuration, don't hesitate to seek further assistance.

**Comment by 237779932** [comment]
@sfmskywalker

I tried, and the child workflow executed successfully, but the last activity of the parent workflow did not execute

![image](https://github.com/elsa-workflows/elsa-core/assets/18566185/02ff88af-87f2-481f-8905-0776223dfd1a)

I debug it, but I can't find the problem.

I hope I can get your help.

**Comment by 237779932** [comment]
![image](https://github.com/elsa-workflows/elsa-core/assets/18566185/bbeae3fe-7b65-44b1-83f3-d4bd5d5d2600)

**Comment by sfmskywalker** [comment]
Hi @237779932 , I just tried your sample workflows in the latest version of Elsa, and for me it works well:

<img width="2560" alt="image" src="https://github.com/elsa-workflows/elsa-core/assets/938393/b8e40f27-e0ef-47a7-b283-154b3f1344e9">


I suspect that perhaps the issue was solved with version 3.0.1, and if not, the upcoming 3.0.2 patch release later today. If you want to try out the preview 3.0.2 release, you can [consume the feedz.io package feed](https://v3.elsaworkflows.io/docs/installation/packages#previews) to try it out and see if it fixes the issue for you as well.

**Comment by 237779932** [comment]
@sfmskywalker 
thank you very much
